Source/Particles/Collision/BinaryCollision/DSMC/DSMCFunc.H

```diff
@@ -163,7 +163,11 @@ public:
                     u1y[I1[i1]] = u1xbuf*std::sin(theta) + u1y[I1[i1]]*std::cos(theta);
 #endif
 
-                    CollisionPairFilter(
+                    const int max_process_count = 4; // Pre-defined value, for performance reasons
+                    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+                        (m_process_count < max_process_count), "Too many scattering processes in DSMC routine."
```

Comment:
I know this is the same message I wrote originally, but in hind-sight maybe it is better to notify the user that this is not a hard limit. Maybe something like:
```suggestion
                        (m_process_count < max_process_count), "Too many scattering processes in DSMC routine (hardcoded to only allow 4). Update the max_process_count value in source code to allow more scattering processes."
```